internal/wavefront/metric/connection_test.go

```diff
@@ -167,10 +168,38 @@ func TestConnection(t *testing.T) {
 		_ = conn.Connect("example.com")
 
 		conn.Send([]metric.Metric{{Name: "some.metric"}})
-		conn.FlushMetrics()
+		conn.Flush()
 
 		require.Equal(t, 1, len(mockSender.SentMetrics))
 	})
+
+	t.Run("handles concurrency", func(t *testing.T) {
+		conn := NewTestConnection(&testhelper.StubSender{})
+
+		const runs = 100_000
```

Comment:
😯👍

Reply:
Yeah, this is a probabilistic test. However, 100k runs seems to drive the probability of triggering a concurrency bug to something pretty close to 100%. This test runs in like 60ms which seems acceptable.